### ISSUE TITLE
[Windows] Path parts in strings replaced with undefined

### ISSUE THREAD

**Issue by jamesplease** [vercel/webpack-asset-relocator-loader#161]
OS: Windows 11
webpack-asset-relocator-loader version:  1.7.2
Node version: 16.15.0

### Configuration

```
 {
    test: /\.(m?js|node)$/,
    parser: { amd: false },
    use: {
      loader: '@vercel/webpack-asset-relocator-loader',
      options: {
        outputAssetBase: 'native_modules',
      },
    },
  },
```

### What is the behavior that I'm seeing

Any string containing paths to the project has that pieceof the string replaced with `undefined`.

### What I would expect to see

Path parts not replaced with `undefined`.

### Example

In this example, the project is located at `C:\Users\example-user\web\example-project`, and a file exists at `.\src\some-path\some-file.txt`

```js
var path = 'C:\\Users\\example-user\\web\\example-project\\src\\some-path\\some-file.txt';
console.log(path);
// => undefinedsome-file.txt
```

### More Information

This is an Electron Forge application. The problem has been narrowed down to this plugin (removing the plugin fixes the issue)

**Comment by TuringMachinegun** [comment]
I am observing the same problem on Windows 11. Same code works fine on Linux. I also tried with an older node version (14.19.2), and an older version of this repo (1.6.0), but no luck.

On my side, it seems like `__webpack_require__.ab` is left undefined in the generated code for requiring the native module:
```
```

I'm not sure if useful information, but: I cloned this repo, ran the tests, and all of them pass.

This issue is also possibly related, or at least shows the same symptoms: https://github.com/rstudio/rstudio/issues/10694

**Comment by jamesplease** [comment]
This comment solved the problem for me:

https://github.com/electron-userland/electron-forge/issues/575#issuecomment-422877001

tl;dr: forge’s config can’t handle every type of native module. Some need to be marked as external.

If you try to get the code in that issue working it may work for you as well.

**Comment by jamesplease** [comment]
I also had to add the externals plugin for builds. Glad you got it working!

**Comment by sibelius** [comment]
is there a workaround for this ?

**Comment by alfredomariamilano** [comment]
This is still an issue that should be easily fixable. I'm trying to debug where it all goes wrong.

**Comment by alfredomariamilano** [comment]
> This is still an issue that should be easily fixable. I'm trying to debug where it all goes wrong.

I still cannot pinpoint exactly where the plugin is failing, but if I import a native node module in the webpack entry, instead of importing from another file, then the `__webpack_require__.ab` snippet is created.

**Comment by pppq** [comment]
The following ticket and Electron Forge's documentation page hints at a solution: 
- https://github.com/electron/forge/issues/2412#issuecomment-1062106849
- https://github.com/electron-forge/electron-forge-docs/pull/76

But it looks like once the assignment for `__webpack_require__.ab` appears, `AssetRelocatorPatch` also needs to be added to `webpack.main.config.js` because only the renderer config includes it by default in `plugins`:
- main config defaults: https://github.com/electron/forge/blob/2199fce3b98fbb23b3d535aa4e4675736102788d/packages/plugin/webpack/src/WebpackConfig.ts#L201
- renderer config defaults: https://github.com/electron/forge/blob/2199fce3b98fbb23b3d535aa4e4675736102788d/packages/plugin/webpack/src/WebpackConfig.ts#L260

Combining the two pieces results in a config similar to the one below:

```js
const relocateLoader = require('@vercel/webpack-asset-relocator-loader');
const AssetRelocatorPatch = require('@electron-forge/plugin-webpack/dist/util/AssetRelocatorPatch');

module.exports = {
  entry: './main.js',
  ...
  plugins: [
    {
      apply (compiler) {
        compiler.hooks.compilation.tap('webpack-asset-relocator-loader', (compilation) => {
          relocateLoader.initAssetCache(compilation, 'native_modules');
        });
      }
    },
    new AssetRelocatorPatch.default(false, false),
  ],
}
```

...however this is also a partial solution as the two boolean constructor parameters are `isProd` and `nodeIntegration`; the first of which is not easy to get hold of from the config.
